insights: Add docs for known oob migration issue (#29485)
sourcegraph/sourcegraph: doc/code_insights/how-tos/Troubleshooting.md

    set state = 'failed'
where id = ?;
```

> ## OOB Migration is not at 100%

Reviewer: ```suggestion
## OOB Migration is not at 100% and is stuck greater than 0%
```

A select few customers will have insights disabled, for which 0% would be expected.

This out-of-band migration is titled: **Migrating insight definitions from settings files to database tables as a last stage to use the GraphQL API.**

> The out-of-band migration shouldn't take more than an hour to complete. (It really shouldn't take more than a few minutes.) So if the progress hasn't reached 100% at this time there's definitely an issue.

Reviewer: Clarified some copy
```suggestion
The out-of-band migration shouldn't take more than an hour to complete. (It really shouldn't take more than a few minutes.) If the progress hasn't reached 100% in this duration some records may be stuck due to missing users or organizations.
```

Author: I did leave this as "stuck due to errors," to be more general, because it's possible there is some other error that could be thrown. It just wouldn't be one that we know about.

Reviewer: I agree, but I think we should be specific that this resolution should only apply to a specific set of errors. Otherwise, we should have the customer engage support in case there is something else going on.

Known issues:
- Deleted users/orgs will cause processing errors, and those jobs wil need to be manually marked as complete.

### Diagnose and Resolve
1. First check the Recent Errors under the migration in the UI.
    1. If the error messages are all: `UserStoreGetById: user not found`
        - This is caused by deleted users. It will be safe to mark these rows as completed by running the following against `pgsql`:
          ```sql

>           UPDATE insights_settings_migration_jobs SET completed_at = NOW() WHERE completed_at IS NULL;

Reviewer: Alternatively, we could specify just the jobs that report errors.

```suggestion
          UPDATE insights_settings_migration_jobs SET completed_at = NOW() WHERE completed_at IS NULL and user_id in ();
```

Author: I'm a bit concerned how they are supposed to get these user ids.. I guess they could manually make a set from the Recent Errors list? That seems sort of error prone, and I'm not sure there's a benefit. It should end up being the same set. What do you think?

Reviewer 2: I think this is safe enough without specifying unique IDs.

Reviewer: I'm fine either way, this could just have the unintended consequence that we reset jobs that aren't complete. I think its low risk and low impact, so this works for me.

          ```

    2. If the error messages are all: `OrgStoreGetByID: org not found`
        - This is caused by deleted orgs. In this case, mark just the org rows as completed by running the following against `pgsql`:
          ```sql

>           UPDATE insights_settings_migration_jobs SET completed_at = NOW() WHERE completed_at IS NULL AND org_id IS NOT NULL;

Reviewer: Same as above
```suggestion
          UPDATE insights_settings_migration_jobs SET completed_at = NOW() WHERE completed_at IS NULL AND org_id in ();
```

Reviewer 2: I think this is safe enough without specifying unique IDs.
